### DISCUSSION TITLE
Phalcon 5 - Problem in Form element attributes

### DISCUSSION THREAD

**Discussion by Alistar84** [phalcon/cphalcon#16122] [Q&A] [answered]
Good morning,
I'm trying to migrate my project from v4 to v5.
I noticed some strange behavior in Form class.

When I create a simple element like this:
`$username = new Text('username', array('required'=>true, 'class'=>'form-control', 'maxlength'=>50));`
I've got an error: Input must be an array or a string.

This happens because the method `Phalcon\Html\Escaper->attributes()` now accepts only string or array.

If I change the code by putting numbers and booleans between quotes, all works fine:
`$username = new Text('username', array('required'=>'true', 'class'=>'form-control', 'maxlength'=>'50'));`

I need to edit all my forms elements, or is this a bug?

**Comment by s-ohnishi** [comment]
Hi.

Since the required attribute value is a boolean
`<input type="text" required ...>`
or
`<input type="text" required="required" ...>`
I think it's normal to write

`<input type="text" required="" ...>` or
`<input type="text" required="1" ...>` would be fine, but don't you think it's disgusting? (I would like to avoid)

And Phalcon seems to adopt a method that does not abbreviate like `required="required"`, so I think it would be best to write it as follows.
`new Text('username', array('required'=>'required', ... ))`

Many attribute values ​​need to specify strings, and if `attribute="attribute"` is allowed as HTML as a boolean specification method, how Phalcon always writes strings as attribute values I think that it is a logical operation to be unified into (= no need for unnecessary branches).

It's very unfortunate that you end up in a situation where you have to rewrite every applicable element, but I support Phalcon's direction.

**Comment by niden** [comment]
From what I read all attributes _should_ be passed as string but it is not enforced 

I will make the necessary changes and release today.

**Comment by niden** [comment, marked as answer]
@Alistar84 You can upgrade to 5.0.2 and that issue is fixed

https://github.com/phalcon/cphalcon/releases/tag/v5.0.2
